src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/BaseSyncStrategy.java

```diff
@@ -164,4 +168,34 @@ void stopSync() {
     abstract void doStart(SyncContext context, int syncParallelism);
 
     abstract void doStop();
+
+    void putSyncRequest(SyncRequest request, AtomicBoolean syncing) {
+        if (!syncing.get()) {
+            logger.atTrace(SYNC_EVENT_TYPE)
+                    .addKeyValue(LOG_THING_NAME_KEY, request.getThingName())
+                    .addKeyValue(LOG_SHADOW_NAME_KEY, request.getShadowName())
+                    .log("Syncing is stopped. Ignoring sync request");
+            return;
+        }
+        try {
+            logger.atInfo(SYNC_EVENT_TYPE)
+                    .addKeyValue(LOG_THING_NAME_KEY, request.getThingName())
+                    .addKeyValue(LOG_SHADOW_NAME_KEY, request.getShadowName())
+                    .log("Adding new sync request");
```

Comment:
nit: should probably be debug here - if they get a ton of sync items this will spam logs